platform/src/abstractions/EducationBenchmarking.Platform.Infrastructure/Cosmos/CollectionService.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.Extensions.Options;
+
+namespace EducationBenchmarking.Platform.Infrastructure.Cosmos;
+
+public interface ICollectionService
+{
+    Task<DataCollection> GetLatestCollection(string dataGroup);
+}
+
+public class CollectionServiceOptions
+{
+    public string ConnectionString { get; set; }
+    public string DatabaseId { get; set; }
+    public string LookupCollectionName { get; set; }
+}
+
+public class CollectionService : CosmosDatabase, ICollectionService
+{
+    private readonly CollectionServiceOptions _options;
+
+    public CollectionService(IOptions<CollectionServiceOptions> options) : base(options.Value.ConnectionString,
+        options.Value.DatabaseId)
+    {
+        _options = options.Value;
+    }
+
+    private async Task<DataCollection[]> GetAllActiveCollections()
+    {
+        return await GetItemEnumerableAsync<DataCollection>(
+                _options.LookupCollectionName,
+                q => q.Where(x => x.Active == "Y"))
+            .ToArrayAsync();
+    }
+
+
+    private async Task<IEnumerable<DataCollection>> GetActiveCollectionsForDataGroup(string dataGroup)
+    {
+        var groups = await GetAllActiveCollections();
+        return groups.Where(c => c.DataGroup == dataGroup)
+            .ToArray();
+    }
+
+    public async Task<DataCollection> GetLatestCollection(string dataGroup)
+    {
+        var activeCollections = await GetActiveCollectionsForDataGroup(dataGroup);
+
+        return activeCollections.MaxBy(o => o.Name.Split('-').First())
```

Comment:
If _Name_ is a string, e.g. "2022-2023" would the _MaxBy_ function still work? I'd assume that the call to _Split_ would return "2022","2023" as string types, not numeric?

Reply:
The naming convention on the collections is  {timestamp}-{type}-{year_from}-{year_to}. 

The spilt is getting the first element which is the timestamp. 

MaxBy function isn't limited to use numberic. It will use build-in comparer (or custom should you so wish) for the type building used.

https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.maxby?view=net-8.0